### PR TITLE
DAOS-11457 test: support multiple launch.py -e

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: pr,-hw
Skip-unit-tests: true
Skip-fault-injection-test: true

Support multiple extra yaml files to launch.py

Required-githooks: true

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

### CODE REVIEW

File: src/tests/ftest/launch.py [hunk 1]
```diff
@@ -747,8 +747,11 @@ def get_test_files(test_list, args, yaml_dir):
 
     """
     # Replace any placeholders in the extra yaml file, if provided
+    combined_extra_yaml = None
     if args.extra_yaml:
-        args.extra_yaml = replace_yaml_file(args.extra_yaml, args, yaml_dir)
+        for extra_yaml in args.extra_yaml:
+            combined_extra_yaml = replace_yaml_file(extra_yaml, args, yaml_dir)
+        args.extra_yaml = combined_extra_yaml
```
Comment (phender) [inline src/tests/ftest/launch.py:754]: args.extra_yaml should retain its original list typing:
```
args.extra_yaml = [replace_yaml_file(extra, args, yaml_dir) for extra in args.extra_yaml]
```

And later on replace lines 764-765 with just:
```
    command.extend(args.extra_yaml)
```

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/launch.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/launch.py:754]: Oh, avocado accepts multiple :)

Comment (daltonbohning) [reply src/tests/ftest/launch.py:754]: This way means each `args.extra_yaml` will essentially contain duplicate base yaml. And then later on, we'll be passing all those:
https://github.com/daos-stack/daos/blob/0b7796cff2eae0cc3a6b9ed0366b1436194a707a/src/tests/ftest/launch.py#L1305-L1306
Functionally, it seems okay, but overkill.
Note `args.extra_yaml` is referenced in both `get_test_files` and `run_tests`.

File: src/tests/ftest/launch.py [hunk 2]
```diff
@@ -759,7 +758,7 @@ def get_test_files(test_list, args, yaml_dir):
         # Display the modified yaml file variants with debug
         command = ["avocado", "variants", "--mux-yaml", test_file["yaml"]]
         if args.extra_yaml:
-            command.append(args.extra_yaml)
+            command.extend(args.extra_yaml)
```
Comment (phender) [inline src/tests/ftest/launch.py:761]: Do we need the conditional before this?

File: src/tests/ftest/launch.py [hunk 3]
```diff
@@ -1300,7 +1299,7 @@ def run_tests(test_files, tag_filter, args):
             test_command_list = list(command_list)
             test_command_list.extend(["--mux-yaml", test_file["yaml"]])
             if args.extra_yaml:
-                test_command_list.append(args.extra_yaml)
+                test_command_list.extend(args.extra_yaml)
```
Comment (phender) [inline src/tests/ftest/launch.py:1302]: Same question?

File: src/tests/ftest/launch.py (hunk 2 above)
Comment (daltonbohning) [reply src/tests/ftest/launch.py:761]: Yes. `ArgumentParser` sets the default as `None`, which is not iterable. Even when I changed the default to `[]`, `ArgumentParser` set it to `None`

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Review by phender — APPROVED** [review]

**Pull request merged**